packages/tools/examples/interpolationContourSegmentation/index.ts

```diff
@@ -0,0 +1,362 @@
+import { Events } from '../../src/enums';
+
+// This is for debugging purposes
+console.warn(
+  'Click on index.ts to open source code for this example --------->'
+);
+
+const {
+  PlanarFreehandContourSegmentationTool,
+  PlanarFreehandROITool,
+  SplineContourSegmentationTool,
+  SplineROITool,
+  LivewireContourSegmentationTool,
+  LivewireContourTool,
+  SegmentationDisplayTool,
+  ToolGroupManager,
+  Enums: csToolsEnums,
+  segmentation,
+} = cornerstoneTools;
+
+const { ViewportType } = Enums;
+const { MouseBindings } = csToolsEnums;
+
+// Define a unique id for the volume
+const volumeName = 'CT_VOLUME_ID'; // Id of the volume less loader prefix
+const volumeLoaderScheme = 'cornerstoneStreamingImageVolume'; // Loader id which defines which volume loader to use
+const volumeId = `${volumeLoaderScheme}:${volumeName}`; // VolumeId with loader id + volume id
+const renderingEngineId = 'myRenderingEngine';
+const viewportIds = ['CT_STACK', 'CT_VOLUME_SAGITTAL'];
+const segmentationId = `SEGMENTATION_ID`;
+let segmentationRepresentationUID;
+
+const interpolationTools = new Map<string, any>();
+const configuration = {
+  interpolation: { enabled: true },
+};
+interpolationTools.set('FreeformInterpolation', {
+  baseTool: PlanarFreehandContourSegmentationTool.toolName,
+  configuration,
+});
+interpolationTools.set('SplineInterpolation', {
+  baseTool: SplineContourSegmentationTool.toolName,
+  configuration,
+});
+interpolationTools.set('LivewireInterpolation', {
+  baseTool: LivewireContourSegmentationTool.toolName,
+  configuration,
+});
+const interpolationToolName = [...interpolationTools.keys()][0];
+
+const segmentIndexes = [1, 2, 3, 4, 5];
+
+// ======== Set up page ======== //
+setTitleAndDescription(
+  'Contour Freehand Annotation Tool',
+  'Here we demonstrate how to use the Contour Freehand Annotation Tool to draw 2D closed ROIs'
+);
+
+const size = '500px';
+const content = document.getElementById('content');
+const viewportGrid = document.createElement('div');
+
+viewportGrid.style.display = 'flex';
+viewportGrid.style.display = 'flex';
+viewportGrid.style.flexDirection = 'row';
+
+const element1 = document.createElement('div');
+const element2 = document.createElement('div');
+element1.style.width = size;
+element1.style.height = size;
+element2.style.width = size;
+element2.style.height = size;
+
+// Disable right click context menu so we can have right click tool
+element1.oncontextmenu = (e) => e.preventDefault();
+element2.oncontextmenu = (e) => e.preventDefault();
+
+viewportGrid.appendChild(element1);
+viewportGrid.appendChild(element2);
+
+content.appendChild(viewportGrid);
+
+addDropdownToToolbar({
+  labelText: 'Segment Index',
+  options: { values: segmentIndexes, defaultValue: segmentIndexes[0] },
+  onSelectedValueChange: (nameAsStringOrNumber) => {
+    updateActiveSegmentIndex(Number(nameAsStringOrNumber));
+  },
+});
+
+const toolsNames = [
+  ...interpolationTools.keys(),
+  PlanarFreehandROITool.toolName,
```

Comment:
Not sure why we need vanilla ROI tools. This example should focus only on segmentation and interpolation IMO

Reply:
Because sometimes one needs to compare the vanilla tool with the interpolated one.  Also, it allows showing additional contours on the same image as they have different behaviour.